monitor/mesos.go

```diff
@@ -109,17 +123,45 @@ func (m *MesosMonitor) SetMesosAgentsInMaintenance(hosts map[string]string) erro
 
 // HasProtectedFrameworksTasks returns true if the mesos agent has any tasks running from any of the
 // protected frameworks.
-func (m *MesosMonitor) HasProtectedFrameworksTasks(ipAddress string) bool {
+func (m *MesosMonitor) HasProtectedFrameworksTasks(ipAddress string) (bool, *mesos.Framework) {
 
 	slaveID := m.mesosCache.slaves[ipAddress].ID
 	fmt.Println(m.mesosCache.slaves)
 	slaveTasks := m.mesosCache.tasks[slaveID]
 	for _, task := range slaveTasks {
-		_, ok := m.mesosCache.frameworks[task.FrameworkID]
+		framework, ok := m.mesosCache.frameworks[task.FrameworkID]
 		if ok {
-			return true
+			return true, &framework
 		}
 	}
 
-	return false
+	return false, nil
+}
+
+// HasProtectedLabelsTasks returns true if the mesos agent has any label tasks running from any of the
+// protected label.
+func (m *MesosMonitor) HasProtectedLabelTasks(ipAddress string) (bool, *mesos.Task) {
+
+	slaveID := m.mesosCache.slaves[ipAddress].ID
+	fmt.Println(m.mesosCache.slaves)
+	slaveTasks := m.mesosCache.tasks[slaveID]
+	for _, task := range slaveTasks {
+		log.Infof("Searching Task %v removed. Deleting it", task)
+		_, ok := m.mesosCache.tasksWithLabelProtected[task.Name]
```

Comment:
you don't need to create a new structure for tasks. You can add a flag to the existing task instance, something like "isProtected". Then, the code would just change to:
`
if m.mesosCache.tasksWithLabelProtected[task.Name].isProtected {
  return true, &task
}
`